rero_ils/modules/permissions.py

```diff
@@ -230,3 +238,57 @@ def delete(cls, user, record):
         if user.is_anonymous:
             return False
         return has_superuser_access()
+
+
+class RecordPermissionPolicy(_RecordPermissionPolicy):
+    """The record base permission policy.
+
+    All permissions are deny by default. This should b compatible with
+    invenio-records-rest and invenio-records-resources.
+    """
+
+    can_search = [Disable()]
+    can_read = [Disable()]
+    can_create = [Disable()]
+    can_update = [Disable()]
+    can_delete = [Disable()]
+
+    # Associated files permissions (which are really bucket permissions)
+    can_read_files = [Disable()]
+    can_update_files = [Disable()]
+
+
+class AllowedByAction(Generator):
+    """Allows if the logged user can perform the given action."""
+
+    def __init__(self, action):
+        """Constructor.
+
+        :param action: string - the action name to allows.
+        """
+        self.action = action
+
+    def needs(self, **kwargs):
+        """Allows the given action.
+
+        :param kwargs: extra arguments
+        :returns: a list of action permission.
+        """
+        return [current_access.actions.get(self.action)]
+
+
+class AllowedByActionSameOrganisation(AllowedByAction):
+    """Allow if the user and the recrod have the same organisation."""
+
+    def excludes(self, record=None, **kwargs):
+        """Excludes if the record and the user has different organisation.
+
+        :param record: Record instance - the given record.
+        :returns: a list of permissions, empty if allows.
+        """
+        if record and (
+            not current_librarian
+            or current_librarian.organisation_pid != record.organisation_pid
```

Comment:
I'm annoyed about `current_librarian` usage. The purpose of this Generator is to restrict record usage to user logged for the same organisation, not check if the user is a librarian or not. A `patron` (current_patron) has also the `organisation_pid` property and could be used in this Generator ? What's your feeling ?

Reply:
Most of the resources are limited to librarian. A generator, allowing the same organisation for librarian and patron is useless. I will change the name of this generator to be more explicit.